### PR TITLE
Add Validation, rspec model tests and solve n+1 

### PR DESCRIPTION
**Pull request by karelvanoordt**
- Add validations:
-- User model:
--- for `name`: must not be blank
--- for `posts_counter` must be an integer greater or equal than zero.
-- Post model:
--- for `title`: must not be blank, and length must not exceed 250 characters.
--- for `comments_counter` and `likes_counter`must be integer greater or equal than zero.
-- Add unit tests for `user` and `post` model methods.
-- Add flash messages for create actions
-- Solved n + 1 problem

### CODE REVIEW

File: spec/models/like_spec.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
-require 'rails_helper'
+# require 'rails_helper'
 
-RSpec.describe Like, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
-end
+# RSpec.describe Like, type: :model do
```
Comment (hiromataba) [inline spec/models/like_spec.rb]: - [x] Please add unit tests for your like model as required 

image

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
-require 'rails_helper'
+# require 'rails_helper'
 
-RSpec.describe Comment, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
-end
+# RSpec.describe Comment, type: :model do
+#   pending "add some examples to (or delete) #{__FILE__}"
+# end
```
Comment (hiromataba) [inline spec/models/comment_spec.rb]: - [x] Please add unit  tests for your comment model as required 

image

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi :wave: :wave: @karelvanoordt , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### To highlight :1st_place_medal: 
- Nice PR title and description :heavy_check_mark: 
- Professional README file :heavy_check_mark:
- Your linters are passing :heavy_check_mark:  

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
Hi @karelvanoordt 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**